// vite-plugin-pug/dist/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, copyDefault, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && (copyDefault || key !== "default"))
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toESM = (module2, isNodeMode) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", !isNodeMode && module2 && module2.__esModule ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};
var __toCommonJS = /* @__PURE__ */ ((cache) => {
  return (module2, temp) => {
    return cache && cache.get(module2) || (temp = __reExport(__markAsModule({}), module2, 1), cache && cache.set(module2, temp), temp);
  };
})(typeof WeakMap !== "undefined" ? /* @__PURE__ */ new WeakMap() : 0);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => src_default,
  pugs: () => pugs
});
var import_path = require("path");
var import_pug = require("pug");
var import_picocolors = __toESM(require("picocolors"));
function pugs(html, pugger, logger) {
  return html.replace(/<pug.+?(file|src)="(.+?)".*?\/.*?>/gi, (_tag, attr, filename) => {
    if (attr === "file" && logger) {
      logger.warn(`${import_picocolors.default.red(`the ${import_picocolors.default.bold(`file`)} attribute is deprecated,`)} ${import_picocolors.default.cyan(`please include ${import_picocolors.default.italic(filename)} with ${import_picocolors.default.bold(`src`)} instead`)}`);
    }
    return pugger(filename);
  });
}
function src_default(options, locals) {
  return {
    name: "vite-plugin-pug",
    handleHotUpdate({ file, server }) {
      if (file.endsWith(".pug")) {
        server.config.logger.info(`${import_picocolors.default.red(`pug\u2019s not hot`)} \u{1F32D} ${import_picocolors.default.cyan(file)}`);
        server.ws.send({
          type: "full-reload"
        });
      }
    },
    transformIndexHtml: {
      transform(html, { server, filename: htmlfile }) {
        return pugs(html, (filename) => {
          const compile = (filepath) => (0, import_pug.compileFile)(filepath, options)(locals);
          if (typeof (options == null ? void 0 : options.localImports) === "function" && options.localImports(htmlfile) || (options == null ? void 0 : options.localImports)) {
            const filedir = htmlfile.replace(/(.*)[\\\/].*\.html$/, "$1");
            const filepath = (0, import_path.join)(filedir, filename);
            return compile(filepath);
          }
          return compile(filename);
        }, server == null ? void 0 : server.config.logger);
      }
    }
  };
}
module.exports = __toCommonJS(src_exports);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  pugs
});
