// vite-plugin-pug/dist/index.mjs
// src/index.ts
import { join } from "path";
import { compileFile } from "pug";
import pc from "picocolors";
function pugs(html, pugger, logger) {
  return html.replace(/<pug.+?(file|src)="(.+?)".*?\/.*?>/gi, (_tag, attr, filename) => {
    if (attr === "file" && logger) {
      logger.warn(`${pc.red(`the ${pc.bold(`file`)} attribute is deprecated,`)} ${pc.cyan(`please include ${pc.italic(filename)} with ${pc.bold(`src`)} instead`)}`);
    }
    return pugger(filename);
  });
}
function src_default(options, locals) {
  return {
    name: "vite-plugin-pug",
    handleHotUpdate({ file, server }) {
      if (file.endsWith(".pug")) {
        server.config.logger.info(`${pc.red(`pug\u2019s not hot`)} \u{1F32D} ${pc.cyan(file)}`);
        server.ws.send({
          type: "full-reload"
        });
      }
    },
    transformIndexHtml: {
      transform(html, { server, filename: htmlfile }) {
        return pugs(html, (filename) => {
          const compile = (filepath) => compileFile(filepath, options)(locals);
          if (typeof (options == null ? void 0 : options.localImports) === "function" && options.localImports(htmlfile) || (options == null ? void 0 : options.localImports)) {
            const filedir = htmlfile.replace(/(.*)[\\\/].*\.html$/, "$1");
            const filepath = join(filedir, filename);
            return compile(filepath);
          }
          return compile(filename);
        }, server == null ? void 0 : server.config.logger);
      }
    }
  };
}
export {
  src_default as default,
  pugs
};
